feature_engine/encoding/decision_tree.py

```diff
@@ -116,9 +116,10 @@ class DecisionTreeEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
 
             -  If 'raise', then unseen categories will raise an error.
 
-            -  If 'encode', unseen categories will be encoded as -1 by the 
+            -  If 'encode', unseen categories will be encoded as -1 by the
```

Comment:
Thanks for the fixes. Here are my thoughts regarding the replacement for unseen categories:

I don't think -1 is a good idea, because we don't really know what the output of the tree is going to be. If it is a probability, then -1 is a safe choice, because it clearly signals that this category was encoded by a different logic.

But if the target is continuous, then -1 might be a common output of the tree, and then we'd be treating this category like perhaps other group of categories that appear quite often in the data.

If the target is binary, perhaps we could leave -1. But if the target is continuous, I think it won't be possible for the API to understand what a would replacement value would be.

So my suggestion is: 

- add a `fill_value` parameter at init, defaults to `None`, where the user can enter the value they wish to use for the encoding (this is the logic currently used in sklearn ordinal encoder).
- then, add logic to have unseen categories replaced by the fill_value when unseen=encode
- if unseen equals raise or ignore, we should treat unseen as we normally do: raise throws an error, ignore i think replaces by nan, or leaves it as it is, I don;t remember any more, we should do whatever ordinaleconder does. 

At init, we need a check that `fill_value` is numeric, and another check to corroborate that `fill_value` is not none when `unseen = encode`.

Then we need checks for the values that `unseen` can take. I don't remember if we can inherit this from a parent class, we should check what ordinal encoder does and try to repeat it for this class.

Regarding the logic to replace the unseen categories:

- if raise, we just leave it to ordinalencoder to handle
- if ignore or encode, we need to add extra logic in this class to have those categories either replaced with nan (or whatever ordinalencoder does, or with the `fill_value`.

And then, we need tests for all the new logic, the tests at init, plus the replacements during transform.

What do you think? Do you think you could work on this?

Thank you!

Reply:
You could also rebase main now so that the tests can pass :)

Reply:
I understand your point! Indeed, I think what you are suggesting would the most flexible way. 
I will work on that this week :)

Reply:
awesome! thank you!

Reply:
Now that I am rereading the messages and the code I  have a question. 
With your phrase
<em>So my suggestion is:
*    add a fill_value parameter at init, defaults to None, where the user can enter the value they wish to use for the encoding (this is the logic currently used in sklearn ordinal encoder).
*   then, add logic to have unseen categories replaced by the fill_value when unseen=encode
*   if unseen equals raise or ignore, we should treat unseen as we normally do: raise throws an error, ignore i think replaces by nan, or leaves it as it is, I don;t remember any more, we should do whatever ordinaleconder does.</em>

You mean to have as a init param **fill_value** and **unseen** in the  `DecisionTreeEncoder`  (to forward them to the ordinal enc) and also in the `OrdinalEncoder`. To add within the  `OrdinalEncoder` class the logic of unseen fill_value ?
 
I am asking that, because it is still not 100% clear to me if you meant the change to happen only in the `DecisionTreeEncoder`. 

After having a look I feel the change must happen also in the `OrdinalEncoder` to have the flexibility for the  `OrdinalEncoder` to fill [unseen values with the passed param instead of always with -1](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/encoding/ordinal.py#L224)


More than a question is a confirmation if what I understood (and tried to explain here) is aligned with what you meant  :)

Reply:
Hi @93lorenzo 

We would not be changing the OrdinalEnocoder in this PR. We would forward only the "raise" to the OrdinalEncoder's unseen, because it has that functionality already, but for "ignore" and "encode" we need to add the logic to the tree encoder. The fill_value will exist only in tree encoder as well.

Reply:
Hey @solegalli ! 
I did the changes! 
Please have a look and let me know if there is something to add/change or discuss :)